src/main/java/uk/gov/hmcts/cmc/claimstore/services/UserService.java

```diff
@@ -10,9 +14,13 @@
 public class UserService {
 
     private IdamApi idamApi;
+    private final AuthTokenGenerator cachedServiceAuthTokenGenerator;
 
-    public UserService(final IdamApi idamApi) {
+    @Autowired
+    public UserService(final IdamApi idamApi,
+                       final AuthTokenGenerator cachedServiceAuthTokenGenerator) {
```

Comment:
Are you sure cached instance will get injected here?

Reply:
it should be cached one, first time it caches call original bean and later it uses from it's cache. tested locally while debug using docker